### ISSUE TITLE
Incorrect locking in Configuration.cxx.

### ISSUE THREAD

**Issue by michalsrb** [TigerVNC/tigervnc#278]
Hi, while looking at the common/rfb/Configuration.cxx code I noticed the use of a `LOCK_CONFIG` macro, which is meant to protect the reads and writes of the configuration. The macro is calling `configLock` function which creates `rfb::Mutex` on first use:

```
#ifdef __RFB_THREADING_IMPL
// On platforms that support Threading, we use Locks to make getData safe
#define LOCK_CONFIG Lock l(*configLock())
rfb::Mutex* configLock_ = 0;
static rfb::Mutex* configLock() {
  if (!configLock_)
    configLock_ = new rfb::Mutex;
  return configLock_;
}
#else
#define LOCK_CONFIG
#endif
```

That however introduces another race condition: two threads could potentially both create and lock two separate locks.

Simple solution would be to just make the `configLock` static variable. Threading seems to be currently used only with winvnc and I am not sure if the WINAPI `InitializeCriticalSection` function would have any problem being called during initialization of static variables. I would guess not.

Cooler solution would be to require c++11 and use the new locking tools from standard library.

**Comment by CendioOssman** [comment]
Threading is also used in the viewer and I have a branch that uses it in all servers. But it's a separate, platform independent implementation. So that really should be merged into a single thing.

**Comment by CendioOssman** [comment]
Btw, that code has tried to avoid spreading threads all over the place in order to avoid having locks everywhere. So perhaps WinVNC could use with some reduction in threads instead.

**Comment by CendioOssman** [comment]
Fixed in 338e73aef8dcd3e7e43e4337071ea9d77568c18b.
